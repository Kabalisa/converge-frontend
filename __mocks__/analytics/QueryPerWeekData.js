import ANALYTICS_MEETING_ROOM_PER_WEEK_DATA from '../../src/graphql/queries/analytics';

const analyticsPerWeekData = [
  {
    request: ANALYTICS_MEETING_ROOM_PER_WEEK_DATA,
    result: {
      data: {
        weeklyDurationsOfMeetings: {
          MeetingsDurationaAnalytics: [
            {
              roomName: 'Tortuga',
              count: 22,
              totalDuration: 1125,
            },
            {
              roomName: 'Accra',
              count: 12,
              totalDuration: 670,
            },
          ],
        },
      },
    },
  },
];

export { analyticsPerWeekData as default };
