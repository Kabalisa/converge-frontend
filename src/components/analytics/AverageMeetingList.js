import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { graphql, compose } from 'react-apollo';
import '../../assets/styles/averageMeetinglist.scss';
import TableHead from '../helpers/TableHead';
import QueryAnalyticsPerMeetingRoom from './AverageMeetingList/QueryAnalyticsPerMeetingRoom';
import Tip from '../commons/Tooltip';
import MEETING_DURATION_ANALYTICS from '../../graphql/queries/analytics';
import Pagination from '../commons/Pagination';
import Overlay from '../commons/Overlay';
import { notFoundIcon } from '../../utils/images/images';
import ErrorIcon from '../commons/ErrorIcon';
import meetingDurationAnalyticsMock from '../../fixtures/meetingDurationAnalytics';

/**
 * Component for the average meeting list
 *
 * @param {array} props
 *
 * @returns {JSX}
 */
export class AverageMeetingList extends Component {
  constructor(props) {
    super(props);
    this.state = {
      analyticsForMeetingsDurations: {
        ...props.data.analyticsForMeetingsDurations,
      },
      isFetching: false,
      currentPage: 1,
      perPage: 5,
    };
  }

  componentWillReceiveProps(props) {
    const { analyticsForMeetingsDurations = {} } = props.data;
    this.setState({
      analyticsForMeetingsDurations,
    });
  }

  componentDidUpdate(prevProps, prevState) {
    const { queryCompleted, data: { error, loading } } = this.props;
    const { analyticsForMeetingsDurations } = this.state;
    if (!error && !loading) {
      queryCompleted('AverageMeetingList');
    }
    if (prevState && prevState.analyticsForMeetingsDurations !== analyticsForMeetingsDurations) {
      const { updateParent } = this.props;
      updateParent('averageMeetingTime', analyticsForMeetingsDurations);
    }
  }

  /**
   * fetches data for the given number of pages
   *
   * @param {number} perPage
   * @param {number} currentPage
   *
   * @returns {void}
   */
  handleData = (perPage, currentPage) => {
    this.setState({ isFetching: true });
    /* istanbul ignore next */
    /* Reasoning: find explicit way of testing configuration options */
    this.props.data
      .fetchMore({
        variables: {
          startDate: this.props.dateValue.validatedStartDate,
          endDate: this.props.dateValue.validatedEndDate,
          page: currentPage,
          perPage,
        },
        updateQuery: (prev, { fetchMoreResult }) => {
          this.setState({
            analyticsForMeetingsDurations: fetchMoreResult.analyticsForMeetingsDurations,
            currentPage,
          });
        },
      })
      .then(() => this.setState({ isFetching: false }))
      .catch(() => this.setState({ isFetching: false }));
  };

  showErrorMessage = message => (
    <div className="average-table-error">
      <div className="error_class">
        <img
          className="error_icon"
          src={notFoundIcon}
          alt="error_icon"
        />
        <b>
          <p className="error_msg">
            {message || 'An error occurred, cannot fetch data'}
          </p>
        </b>
      </div>
    </div>
  );
  render() {
    const tip =
     'The number of meetings in a room,  the average number of attendees to these meetings as well as the average duration of the meetings.';
    /* eslint no-param-reassign: "error" */
    const {
      analyticsForMeetingsDurations,
      isFetching,
      perPage,
      currentPage,
    } = this.state;
    const { loading, error } = this.props.data;
    const { isFutureDateSelected } = this.props.dateValue;
    if (loading) {
      analyticsForMeetingsDurations.MeetingsDurationaAnalytics = meetingDurationAnalyticsMock(3);
    }
    return (
      <div className="average-meeting overlay-container">
        {loading && <Overlay />}
        <div className="average-meeting-control">
          <h4 className="header-title">Average time spent/Meeting Room</h4>
          <span className="moreVerticalIcon">{Tip(tip)}</span>
        </div>
        <div className="average-meeting-list">
          {isFetching ? <Overlay id="average-meeting" /> : null}
          <div>
            <TableHead
              titles={['Room', 'No. of meetings', 'Average Meeting Duration']}
            />
            <div>
              {
                isFutureDateSelected ?
                this.showErrorMessage('You cannot fetch data beyond today')
                : (
                error ? (
                  <ErrorIcon
                    message={error.graphQLErrors.length > 0 && 'No resource found'}
                  />
                )
                : <QueryAnalyticsPerMeetingRoom
                  data={analyticsForMeetingsDurations}
                />
                )
              }
            </div>
          </div>
        </div>
        {!loading &&
          <div className="average-meeting-pagination">
            <div>
              { !isFutureDateSelected && !error && <Pagination
                totalPages={analyticsForMeetingsDurations.pages}
                hasNext={analyticsForMeetingsDurations.hasNext}
                hasPrevious={analyticsForMeetingsDurations.hasPrevious}
                handleData={this.handleData}
                isFetching={isFetching}
                perPage={perPage}
                currentPage={currentPage}
              />
            }
            </div>
          </div>
        }
      </div>
    );
  }
}

AverageMeetingList.propTypes = {
  dateValue: PropTypes.shape({
    validatedStartDate: PropTypes.string,
    validatedEndDate: PropTypes.string,
    isFutureDateSelected: PropTypes.bool.isRequired,
  }),
  data: PropTypes.shape({
    analyticsForMeetingsDurations: PropTypes.object,
    fetchMore: PropTypes.func,
    loading: PropTypes.bool,
    error: PropTypes.any,
  }).isRequired,
  queryCompleted: PropTypes.func.isRequired,
  updateParent: PropTypes.func,
};

AverageMeetingList.defaultProps = {
  dateValue: {},
  updateParent: null,
};

export default compose(
  graphql(MEETING_DURATION_ANALYTICS, {
    name: 'data',
    options: props => ({
      variables: {
        startDate: props.dateValue.validatedStartDate,
        endDate: props.dateValue.validatedEndDate,
        page: 1,
        perPage: 5,
      },
    }),
  }),
)(AverageMeetingList);
