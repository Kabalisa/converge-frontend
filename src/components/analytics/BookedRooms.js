/* eslint-disable react/no-array-index-key */
import React from 'react';
import PropTypes from 'prop-types';
import ProgressBar from 'react-toolbox/lib/progress_bar';
import TableHead from '../helpers/TableHead';
import '../../assets/styles/bookedroom.scss';
import Tip from '../commons/Tooltip';
import ErrorIcon from '../commons/ErrorIcon';

/**
 *
 * Component for showing booked rooms
 *
 * @param {Object} bookedRoomsObject
 *
 * @returns {JSX}
 */
const BookedRooms = ({
  pollIcon, bookedRoomText, bookedRoomsList, fetching, error, tip,
}) => (
  <div className="wrap-booked-room">
    <div className="booked-room-header">
      <img src={pollIcon} alt="Pull" />
      <h4>{bookedRoomText}</h4>
      {Tip(tip)}
    </div>
    <div className="booked-room-list">
      <div className="table">
        <TableHead titles={['Room', 'Meetings', '% Share of All Meetings']} />
        {error !== null && Object.values(error).length > 0 ? (
          <div className="table__body">
            <div className="table__row--analytics">
              <ErrorIcon />
            </div>
          </div>
        ) : (
          <div className="table__body">
            {!fetching ? (
              bookedRoomsList.map((room, index) => (
                <div className="table__row--analytics" key={index}>
                  <span>{room.roomName}</span>
                  <span>{room.meetings}</span>
                  <span>{room.percentage}%</span>
                </div>
              ))
            ) : (
              <div className="table__row--loading">
                <ProgressBar type="linear" mode="indeterminate" />
              </div>
            )}
          </div>
        )}
      </div>
    </div>
  </div>
);

BookedRooms.propTypes = {
  pollIcon: PropTypes.string.isRequired,
  bookedRoomText: PropTypes.string.isRequired,
  tip: PropTypes.string.isRequired,
  bookedRoomsList: PropTypes.instanceOf(Array).isRequired,
  fetching: PropTypes.bool.isRequired,
  error: PropTypes.shape({
    error: PropTypes.string,
  }),
};

BookedRooms.defaultProps = {
  error: {},
};

export default BookedRooms;
