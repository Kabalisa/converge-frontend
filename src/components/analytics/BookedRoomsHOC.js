import React from 'react';
import { PropTypes } from 'prop-types';
import PollIcon from '../../assets/images/poll.svg';
import RedPollIcon from '../../assets/images/poll_red.svg';
import { getMostUsedAndLeastUsedRooms } from '../../json_requests';

/**
 * Component for showing booked rooms HOC
 *
 * @param {array} props
 *
 * @returns {JSX}
 */
export class ComposedBooked extends React.Component {
  state = {
    leastUsedRooms: [],
    mostUsedRooms: [],
    fetching: false,
    // eslint-disable-next-line
    error: null
  };

  componentDidMount() {
    this.fetchMostAndLeastUsedRooms(this.props.date);
  }

  componentWillReceiveProps(prevProps) {
    if (this.props.date.startDate !== prevProps.date.startDate) {
      this.fetchMostAndLeastUsedRooms(prevProps.date);
    }
  }

  /**
   * Fetches most and least used rooms
   *
   * @param {date} date
   *
   * @returns {Function}
   */
  fetchMostAndLeastUsedRooms = (date) => {
    this.setState({ fetching: true });
    const { startDate, endDate } = date;

    return (
      getMostUsedAndLeastUsedRooms(startDate, endDate)
        .then((response) => {
          const rooms = response.data['Least Used Rooms'].Room;
          const meetings = response.data['Least Used Rooms'].Meetings;
          const meetingShares =
            response.data['Least Used Rooms']['% Share of All Meetings'];

          const mostUsedRooms = response.data['Most Used Rooms'].Room;
          const mostUsedMeetings = response.data['Most Used Rooms'].Meetings;
          const mostUsedMeetingShares =
            response.data['Most Used Rooms']['% Share of All Meetings'];

          this.setState({
            leastUsedRooms: [rooms, meetings, meetingShares],
            mostUsedRooms: [
              mostUsedRooms,
              mostUsedMeetings,
              mostUsedMeetingShares,
            ],
            fetching: false,
          });
        })
        // this fetching value is not set to false to avoid props validation
        // failure in the child component, otherwise it should be set to false
        // eslint-disable-next-line
        .catch(error => this.setState({ fetching: false, error: error }))
    );
  };

  render() {
    const ComposedBookedRooms = this.props.component;
    if (this.props.bookedRoomText === 'Most Booked Rooms') {
      return (
        <ComposedBookedRooms
          pollIcon={PollIcon}
          tip="The highest number of times meeting rooms were booked in a set time period"
          bookedRoomText={this.props.bookedRoomText}
          fetching={this.state.fetching}
          error={this.state.error}
          bookedRoomsList={this.state.mostUsedRooms}
        />
      );
    }
    return (
      <ComposedBookedRooms
        pollIcon={RedPollIcon}
        tip="The least number of times meeting rooms were booked in a set time period"
        bookedRoomText={this.props.bookedRoomText}
        fetching={this.state.fetching}
        error={this.state.error}
        bookedRoomsList={this.state.leastUsedRooms}
      />
    );
  }
}

ComposedBooked.propTypes = {
  date: PropTypes.shape({
    startDate: PropTypes.string,
    endDate: PropTypes.string,
  }),
  component: PropTypes.func.isRequired,
  bookedRoomText: PropTypes.string.isRequired,
};

ComposedBooked.defaultProps = {
  date: {},
};

export { ComposedBooked as default, getMostUsedAndLeastUsedRooms };
