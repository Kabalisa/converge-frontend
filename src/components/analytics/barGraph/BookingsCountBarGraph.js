import React from 'react';
import PropTypes from 'prop-types';

import QueryBookingsCount from '../QueryBookingsCount';
import Tip from '../../commons/Tooltip';

// import styles
import '../../../../src/assets/styles/barGraphBaseStyle.scss';

/**
 * Bar graph for bookings count
 *
 * @param {Object} dateValueObject
 *
 * @returns {JSX}
 */
const BookingsCountBarGraph = ({ dateValue }) => {
  const tip = 'Total count of bookings in a given time';
  return (
    <article className="bar-graph">
      <section className="graph-header">
        <p className="graph-title">Total Bookings Count</p>
        {Tip(tip)}
      </section>
      <section className="graph-content">
        <QueryBookingsCount dateValue={dateValue} />
      </section>
    </article>
  );
};

BookingsCountBarGraph.propTypes = {
  dateValue: PropTypes.shape({
    startDate: PropTypes.string,
    endDate: PropTypes.string,
  }),
};

BookingsCountBarGraph.defaultProps = {
  dateValue: {},
};

export default BookingsCountBarGraph;
