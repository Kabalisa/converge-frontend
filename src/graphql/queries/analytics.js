import gql from 'graphql-tag';

const MEETING_DURATION_ANALYTICS = gql`
  query durationOfMeetingsPerMeetingRoom(
    $startDate: String!
    $endDate: String!
    $page: Int
    $perPage: Int
  ) {
    analyticsForMeetingsDurations(
      startDate: $startDate
      endDate: $endDate
      page: $page
      perPage: $perPage
    ) {
      hasPrevious
      hasNext
      pages
      MeetingsDurationaAnalytics {
        roomName
        count
        totalDuration
      }
    }
  }
`;

const ANALYTICS_BOOKINGS_COUNT = gql`
  query AnalyticsBookingsCount($startDate: String!, $endDate: String!) {
    bookingsAnalyticsCount(startDate: $startDate, endDate: $endDate) {
      period
      bookings
    }
  }
`;

const CHECKINS_BOOKINGS_CANCELLATIONS_PERCENTAGES = gql`
  query AnalyticsRatiosTotal($startDate: String!, $endDate: String!) {
    analyticsRatios(startDate: $startDate, endDate: $endDate) {
      checkins
      checkinsPercentage
      bookings
      cancellations
      cancellationsPercentage
    }
  }
`;

const MOST_BOOKED_ROOMS_ANALYTICS = gql`
  query analyticsForMostBookedRooms($startDate: String!, $endDate: String!) {
    analyticsForBookedRooms(
      startDate: $startDate
      endDate: $endDate
      limit: 10
      criteria: "most_booked"
    ) {
      analytics {
        roomName
        meetings
        percentage
      }
    }
  }
`;

const LEAST_BOOKED_ROOMS_ANALYTICS = gql`
  query analyticsForLeastBookedRooms($startDate: String!, $endDate: String!) {
    analyticsForBookedRooms(
      startDate: $startDate
      endDate: $endDate
      limit: 10
      criteria: "least_booked"
    ) {
      analytics {
        roomName
        meetings
        percentage
      }
    }
  }
`;

const ANALYTICS_FOR_DAILY_ROOM_EVENTS = gql`
  query analyticsForDailyRoomEvents($startDate: String!, $endDate: String!) {
    analyticsForDailyRoomEvents(startDate: $startDate, endDate: $endDate) {
      DailyRoomEvents {
        day
        events {
          eventSummary
          startTime
          endTime
          eventId
          noOfParticipants
          roomName
        }
      }
    }
  }
`;

export {
  MEETING_DURATION_ANALYTICS as default,
  ANALYTICS_BOOKINGS_COUNT,
  CHECKINS_BOOKINGS_CANCELLATIONS_PERCENTAGES,
  MOST_BOOKED_ROOMS_ANALYTICS,
  LEAST_BOOKED_ROOMS_ANALYTICS,
  ANALYTICS_FOR_DAILY_ROOM_EVENTS,
};
