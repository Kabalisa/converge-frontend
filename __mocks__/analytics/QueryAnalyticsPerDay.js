import { ANALYTICS_MEETING_ROOM_PER_DAY } from '../../src/graphql/queries/analytics';

const queryPerDayData = [
  {
    request: ANALYTICS_MEETING_ROOM_PER_DAY,
    result: {
      data: {
        dailyDurationsOfMeetings: {
          MeetingsDurationaAnalytics: [
            {
              roomName: 'Tortuga',
              count: 45,
              totalDuration: 4567,
            },
            {
              roomName: 'Accra',
              count: 19,
              totalDuration: 8907,
            },
          ],
        },
      },
    },
  },
];
export { queryPerDayData as default };
