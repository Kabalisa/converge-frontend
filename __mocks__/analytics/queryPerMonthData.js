import { ANALYTICS_MEETING_ROOM_PER_MONTH } from '../../src/graphql/queries/analytics';

const queryPerMonthData = [
  {
    request: ANALYTICS_MEETING_ROOM_PER_MONTH,
    result: {
      data: {
        monthlyDurationsOfMeetings: {
          MeetingsDurationaAnalytics: [
            {
              roomName: 'Tortuga',
              count: 78,
              totalDuration: 5547,
            },
            {
              roomName: 'Accra',
              count: 73,
              totalDuration: 3313,
            },
          ],
        },
      },
    },
  },
];


export { queryPerMonthData as default };
