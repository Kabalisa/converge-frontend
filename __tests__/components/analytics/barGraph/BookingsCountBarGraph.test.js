import React from 'react';
import { mount } from 'enzyme';
import { MockedProvider } from 'react-apollo/test-utils';
import BookingsCountBarGraph from '../../../../src/components/analytics/barGraph/BookingsCountBarGraph';
import { queryBookingsCountMockData } from '../../../../__mocks__/analytics/queryBookingsCountMockData';

const props = {
  dateValue: { startDate: 'Nov 01 2018', endDate: 'Nov 03 2018' },
};

const analyticsMocks = queryBookingsCountMockData;

describe('Bookings Count Bar Graph Component', () => {
  const wrapper = mount(
    <MockedProvider mocks={analyticsMocks}>
      <BookingsCountBarGraph {...props} />
    </MockedProvider>,
  );
  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('should be able to hover', () => {
    const helpIcon = wrapper.find('img[alt="help icon"]');
    const nativeEvent = { nativeEvent: { clientX: 300, clientY: 250 } };
    helpIcon.simulate('mouseenter', nativeEvent);
    expect(wrapper).toMatchSnapshot();
    helpIcon.simulate('mouseleave', nativeEvent);
    expect(wrapper).toMatchSnapshot();
  });
});
