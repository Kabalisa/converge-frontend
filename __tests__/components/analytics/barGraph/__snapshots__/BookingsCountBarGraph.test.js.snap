// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bookings Count Bar Graph Component renders correctly 1`] = `
<MockedProvider
  addTypename={true}
  mocks={Array []}
>
  <ApolloProvider
    client={
      ApolloClient {
        "__operations_cache__": Map {},
        "cache": InMemoryCache {
          "addTypename": true,
          "config": Object {
            "addTypename": true,
            "dataIdFromObject": [Function],
            "fragmentMatcher": HeuristicFragmentMatcher {},
            "storeFactory": [Function],
          },
          "data": ObjectCache {
            "data": Object {},
          },
          "optimistic": Array [],
          "silenceBroadcast": false,
          "watches": Array [],
        },
        "defaultOptions": Object {},
        "disableNetworkFetches": false,
        "link": ApolloLink {
          "request": [Function],
        },
        "mutate": [Function],
        "query": [Function],
        "queryDeduplication": true,
        "reFetchObservableQueries": [Function],
        "resetStore": [Function],
        "resetStoreCallbacks": Array [],
        "ssrMode": false,
        "store": DataStore {
          "cache": InMemoryCache {
            "addTypename": true,
            "config": Object {
              "addTypename": true,
              "dataIdFromObject": [Function],
              "fragmentMatcher": HeuristicFragmentMatcher {},
              "storeFactory": [Function],
            },
            "data": ObjectCache {
              "data": Object {},
            },
            "optimistic": Array [],
            "silenceBroadcast": false,
            "watches": Array [],
          },
        },
        "version": "2.3.1",
        "watchQuery": [Function],
      }
    }
  >
    <BookingsCountBarGraph
      dateValue={
        Object {
          "endDate": "Nov 03 2018",
          "startDate": "Nov 01 2018",
        }
      }
      updateParent={[MockFunction]}
    >
      <article
        className="bar-graph"
      >
        <section
          className="graph-header"
        >
          <p
            className="graph-title"
          >
            Total Bookings Count
          </p>
          <ThemedTooltippedComponent
            className=""
            composeTheme="deeply"
            mapThemrProps={[Function]}
            theme={null}
            tooltip="Total count of bookings in a given time"
            tooltipDelay={0}
            tooltipHideOnClick={true}
            tooltipPosition="top"
            tooltipShowOnClick={true}
          >
            <TooltippedComponent
              className=""
              theme={
                Object {
                  "tooltipInner": "tooltipInner",
                }
              }
              tooltip="Total count of bookings in a given time"
              tooltipDelay={0}
              tooltipHideOnClick={true}
              tooltipPosition="top"
              tooltipShowOnClick={true}
            >
              <ThemedLink
                active={false}
                className=""
                composeTheme="deeply"
                mapThemrProps={[Function]}
                onClick={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                theme={
                  Object {
                    "tooltipInner": "tooltipInner",
                  }
                }
              >
                <Link
                  active={false}
                  className=""
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  theme={
                    Object {
                      "tooltipInner": "tooltipInner",
                    }
                  }
                >
                  <a
                    className=""
                    data-react-toolbox="link"
                    onClick={[Function]}
                    onMouseEnter={[Function]}
                    onMouseLeave={[Function]}
                  >
                    <img
                      alt="help icon"
                      src="https://firebasestorage.googleapis.com/v0/b/learning-map-app.appspot.com/o/MrmFrontImages%2Fhelp_outline_24px.svg?alt=media&token=839e3137-c29e-463d-9a68-361e28794911"
                    />
                  </a>
                </Link>
              </ThemedLink>
            </TooltippedComponent>
          </ThemedTooltippedComponent>
        </section>
        <section
          className="graph-content"
        >
          <QueryBookingsCount
            dateValue={
              Object {
                "endDate": "Nov 03 2018",
                "startDate": "Nov 01 2018",
              }
            }
            updateParent={[MockFunction]}
          >
            <Spinner
              size=""
            >
              <div
                className="centered"
              >
                <div
                  className="spinner"
                />
              </div>
            </Spinner>
          </QueryBookingsCount>
        </section>
      </article>
    </BookingsCountBarGraph>
  </ApolloProvider>
</MockedProvider>
`;

exports[`Bookings Count Bar Graph Component should be able to hover 1`] = `
<MockedProvider
  addTypename={true}
  mocks={Array []}
>
  <ApolloProvider
    client={
      ApolloClient {
        "__operations_cache__": Map {},
        "cache": InMemoryCache {
          "addTypename": true,
          "config": Object {
            "addTypename": true,
            "dataIdFromObject": [Function],
            "fragmentMatcher": HeuristicFragmentMatcher {},
            "storeFactory": [Function],
          },
          "data": ObjectCache {
            "data": Object {},
          },
          "optimistic": Array [],
          "silenceBroadcast": false,
          "watches": Array [],
        },
        "defaultOptions": Object {},
        "disableNetworkFetches": false,
        "link": ApolloLink {
          "request": [Function],
        },
        "mutate": [Function],
        "query": [Function],
        "queryDeduplication": true,
        "reFetchObservableQueries": [Function],
        "resetStore": [Function],
        "resetStoreCallbacks": Array [],
        "ssrMode": false,
        "store": DataStore {
          "cache": InMemoryCache {
            "addTypename": true,
            "config": Object {
              "addTypename": true,
              "dataIdFromObject": [Function],
              "fragmentMatcher": HeuristicFragmentMatcher {},
              "storeFactory": [Function],
            },
            "data": ObjectCache {
              "data": Object {},
            },
            "optimistic": Array [],
            "silenceBroadcast": false,
            "watches": Array [],
          },
        },
        "version": "2.3.1",
        "watchQuery": [Function],
      }
    }
  >
    <BookingsCountBarGraph
      dateValue={
        Object {
          "endDate": "Nov 03 2018",
          "startDate": "Nov 01 2018",
        }
      }
      updateParent={[MockFunction]}
    >
      <article
        className="bar-graph"
      >
        <section
          className="graph-header"
        >
          <p
            className="graph-title"
          >
            Total Bookings Count
          </p>
          <ThemedTooltippedComponent
            className=""
            composeTheme="deeply"
            mapThemrProps={[Function]}
            theme={null}
            tooltip="Total count of bookings in a given time"
            tooltipDelay={0}
            tooltipHideOnClick={true}
            tooltipPosition="top"
            tooltipShowOnClick={true}
          >
            <TooltippedComponent
              className=""
              theme={
                Object {
                  "tooltipInner": "tooltipInner",
                }
              }
              tooltip="Total count of bookings in a given time"
              tooltipDelay={0}
              tooltipHideOnClick={true}
              tooltipPosition="top"
              tooltipShowOnClick={true}
            >
              <ThemedLink
                active={false}
                className=""
                composeTheme="deeply"
                mapThemrProps={[Function]}
                onClick={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                theme={
                  Object {
                    "tooltipInner": "tooltipInner",
                  }
                }
              >
                <Link
                  active={false}
                  className=""
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  theme={
                    Object {
                      "tooltipInner": "tooltipInner",
                    }
                  }
                >
                  <a
                    className=""
                    data-react-toolbox="link"
                    onClick={[Function]}
                    onMouseEnter={[Function]}
                    onMouseLeave={[Function]}
                  >
                    <img
                      alt="help icon"
                      src="https://firebasestorage.googleapis.com/v0/b/learning-map-app.appspot.com/o/MrmFrontImages%2Fhelp_outline_24px.svg?alt=media&token=839e3137-c29e-463d-9a68-361e28794911"
                    />
                  </a>
                </Link>
              </ThemedLink>
            </TooltippedComponent>
          </ThemedTooltippedComponent>
        </section>
        <section
          className="graph-content"
        >
          <QueryBookingsCount
            dateValue={
              Object {
                "endDate": "Nov 03 2018",
                "startDate": "Nov 01 2018",
              }
            }
            updateParent={[MockFunction]}
          >
            <Spinner
              size=""
            >
              <div
                className="centered"
              >
                <div
                  className="spinner"
                />
              </div>
            </Spinner>
          </QueryBookingsCount>
        </section>
      </article>
    </BookingsCountBarGraph>
  </ApolloProvider>
</MockedProvider>
`;

exports[`Bookings Count Bar Graph Component should be able to hover 2`] = `
<MockedProvider
  addTypename={true}
  mocks={Array []}
>
  <ApolloProvider
    client={
      ApolloClient {
        "__operations_cache__": Map {},
        "cache": InMemoryCache {
          "addTypename": true,
          "config": Object {
            "addTypename": true,
            "dataIdFromObject": [Function],
            "fragmentMatcher": HeuristicFragmentMatcher {},
            "storeFactory": [Function],
          },
          "data": ObjectCache {
            "data": Object {},
          },
          "optimistic": Array [],
          "silenceBroadcast": false,
          "watches": Array [],
        },
        "defaultOptions": Object {},
        "disableNetworkFetches": false,
        "link": ApolloLink {
          "request": [Function],
        },
        "mutate": [Function],
        "query": [Function],
        "queryDeduplication": true,
        "reFetchObservableQueries": [Function],
        "resetStore": [Function],
        "resetStoreCallbacks": Array [],
        "ssrMode": false,
        "store": DataStore {
          "cache": InMemoryCache {
            "addTypename": true,
            "config": Object {
              "addTypename": true,
              "dataIdFromObject": [Function],
              "fragmentMatcher": HeuristicFragmentMatcher {},
              "storeFactory": [Function],
            },
            "data": ObjectCache {
              "data": Object {},
            },
            "optimistic": Array [],
            "silenceBroadcast": false,
            "watches": Array [],
          },
        },
        "version": "2.3.1",
        "watchQuery": [Function],
      }
    }
  >
    <BookingsCountBarGraph
      dateValue={
        Object {
          "endDate": "Nov 03 2018",
          "startDate": "Nov 01 2018",
        }
      }
      updateParent={[MockFunction]}
    >
      <article
        className="bar-graph"
      >
        <section
          className="graph-header"
        >
          <p
            className="graph-title"
          >
            Total Bookings Count
          </p>
          <ThemedTooltippedComponent
            className=""
            composeTheme="deeply"
            mapThemrProps={[Function]}
            theme={null}
            tooltip="Total count of bookings in a given time"
            tooltipDelay={0}
            tooltipHideOnClick={true}
            tooltipPosition="top"
            tooltipShowOnClick={true}
          >
            <TooltippedComponent
              className=""
              theme={
                Object {
                  "tooltipInner": "tooltipInner",
                }
              }
              tooltip="Total count of bookings in a given time"
              tooltipDelay={0}
              tooltipHideOnClick={true}
              tooltipPosition="top"
              tooltipShowOnClick={true}
            >
              <ThemedLink
                active={false}
                className=""
                composeTheme="deeply"
                mapThemrProps={[Function]}
                onClick={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                theme={
                  Object {
                    "tooltipInner": "tooltipInner",
                  }
                }
              >
                <Link
                  active={false}
                  className=""
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  theme={
                    Object {
                      "tooltipInner": "tooltipInner",
                    }
                  }
                >
                  <a
                    className=""
                    data-react-toolbox="link"
                    onClick={[Function]}
                    onMouseEnter={[Function]}
                    onMouseLeave={[Function]}
                  >
                    <img
                      alt="help icon"
                      src="https://firebasestorage.googleapis.com/v0/b/learning-map-app.appspot.com/o/MrmFrontImages%2Fhelp_outline_24px.svg?alt=media&token=839e3137-c29e-463d-9a68-361e28794911"
                    />
                  </a>
                </Link>
              </ThemedLink>
            </TooltippedComponent>
          </ThemedTooltippedComponent>
        </section>
        <section
          className="graph-content"
        >
          <QueryBookingsCount
            dateValue={
              Object {
                "endDate": "Nov 03 2018",
                "startDate": "Nov 01 2018",
              }
            }
            updateParent={[MockFunction]}
          >
            <Spinner
              size=""
            >
              <div
                className="centered"
              >
                <div
                  className="spinner"
                />
              </div>
            </Spinner>
          </QueryBookingsCount>
        </section>
      </article>
    </BookingsCountBarGraph>
  </ApolloProvider>
</MockedProvider>
`;
